daemon/api.go

```diff
@@ -1294,3 +1303,77 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
 
 	return SyncResponse(createResponseData, nil)
 }
+
+// signNonceWithHelper invokes "snap-identity-helper sign-device-nonce <nonce>"
+// and passes a device-key ("openpgp <base64-encoded data>") into stdin.
+// The helper is expected to sign the nonce with the corresponding
+// private key and write the encoded signature ("openpgp [...]") to
+// stdout.
+// TODO: Needs to be overrideable by gadget snap.
+func signNonceWithHelper(deviceKeyEncoded []byte, nonce []byte) ([]byte, error) {
+	cmd := exec.Command("snap-identity-helper", "sign-device-nonce", string(nonce))
+	stdin, err := cmd.StdinPipe()
+	if _, err := stdin.Write(deviceKeyEncoded); err != nil {
+		return nil, err
+	}
+	stdin.Close()
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+	return output, nil
+}
+
+type deviceIdentityChange struct {
+	Brand           string `json:"brand"`
+	Model           string `json:"model"`
+	Serial          string `json:"serial"`
```

Comment:
This data comprises the serial assertion. Do we need to take it both individually and as an assertion?